pkg/storage/chunk/cache/cache.go

```diff
@@ -113,7 +113,11 @@ func New(cfg Config, reg prometheus.Registerer, logger log.Logger) (Cache, error
 			cfg.Redis.Expiration = cfg.DefaultValidity
 		}
 		cacheName := cfg.Prefix + "redis"
-		cache := NewRedisCache(cacheName, NewRedisClient(&cfg.Redis), logger)
+		client, err := NewRedisClient(&cfg.Redis)
+		if err != nil {
+			return nil, err
```

Comment:
nit: Can we wrap the error with some context here?
```suggestion
			return nil, fmt.Errorf("redis client setup failed: %w", err)
```